torch/distributed/rpc/api.py

```diff
@@ -350,6 +351,25 @@ def _validate_rpc_args(backend, store, name, rank, world_size, rpc_backend_optio
             )
 
 
+T = TypeVar("T")
+GenericWithOneTypeVar = Generic[T]
+
+
+try:
+    class RRef(PyRRef, GenericWithOneTypeVar):
+        # Combine the implementation class and the type class.
+        pass
+except TypeError as exc:
+    # TypeError: metaclass conflict: the metaclass of a derived class
+    # must be a (non-strict) subclass of the metaclasses of all its bases
+    class RRefMeta(PyRRef.__class__, GenericWithOneTypeVar.__class__):
```

Comment:
As the previous comment, I think we can safely remove those if this is just for python 3.5

Reply:
@wanchaol XLA tests are still raising this error.

Reply:
maybe bc xla still use 3.5... let's leave it as is then.

Reply:
Shall we mention this is for PY35 only? And shall we add a pointer to the explanation of why we are getting this error?

Reply:
I used to be able to reproduce this exception in py 3.5 on fb internal platform. But I just checked all pytorch CI runs are in python 3.6 now. I can't see a clear reason why for some tests it still raises this exception. I noticed before that pybind11 version play a role here. But I haven't verify the hypothesis yet.